File: BE/routes/saleRouter.js
const express=require("express")
const { getAllSales,addSale,deleteSale,updateSale, getSaleById } = require("../controllers/SaleController")
const verifyJWT = require("../middlewares/verifyJWT")
const router=express.Router()
router.get('/',getAllSales)
router.post('/',addSale)
router.delete('/:id',verifyJWT,deleteSale)
router.put('/:id',verifyJWT,updateSale)
router.get('/:id',verifyJWT,getSaleById)
module.exports=router

File: BE/routes/saleRouter.py
from fastapi import APIRouter, Depends
from controllers.saleController import (
    get_all_sales,
    add_sale,
    delete_sale,
    update_sale,
    get_sale_by_id
)
from middlewares.verifyJWT import verify_jwt

router = APIRouter(
    prefix="/sales",
    tags=["sales"]
)

@router.get("/")
async def route_get_all_sales():
    return await get_all_sales()

@router.post("/")
async def route_add_sale():
    return await add_sale()

@router.delete("/{id}")
async def route_delete_sale(id: str, user=Depends(verify_jwt)):
    return await delete_sale(id)

@router.put("/{id}")
async def route_update_sale(id: str, user=Depends(verify_jwt)):
    return await update_sale(id)

@router.get("/{id}")
async def route_get_sale_by_id(id: str, user=Depends(verify_jwt)):
    return await get_sale_by_id(id)